contracts/lootbox-v1/contracts/Library/AuthenticatedProxy.sol

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.6.2;
+
+contract AuthenticatedProxy is TokenRecipient, OwnedUpgradeabilityStorage {
+    /* Whether initialized. */
+    bool initialized = false;
+
+    /* Address which owns this proxy. */
+    address public user;
+
+    /* Associated registry with contract authentication information. */
+    ProxyRegistry public registry;
+
+    /* Whether access has been revoked. */
+    bool public revoked;
+
+    /* Delegate call could be used to atomically transfer multiple assets owned by the proxy contract with one order. */
+    enum HowToCall {
+        Call,
+        DelegateCall
+    }
+
+    /* Event fired when the proxy access is revoked or unrevoked. */
+    event Revoked(bool revoked);
+
+    /**
+     * Initialize an AuthenticatedProxy
+     *
+     * @param addrUser Address of user on whose behalf this proxy will act
+     * @param addrRegistry Address of ProxyRegistry contract which will manage this proxy
+     */
+    function initialize(address addrUser, ProxyRegistry addrRegistry) public {
+        require(!initialized);
+        initialized = true;
+        user = addrUser;
+        registry = addrRegistry;
+    }
+
+    /**
+     * Set the revoked flag (allows a user to revoke ProxyRegistry access)
+     *
+     * @dev Can be called by the user only
+     * @param revoke Whether or not to revoke access
+     */
+    function setRevoke(bool revoke) public {
+        require(msg.sender == user);
+        revoked = revoke;
+        emit Revoked(revoke);
+    }
+
+    /**
+     * Execute a message call from the proxy contract
+     *
+     * @dev Can be called by the user, or by a contract authorized by the registry as long as the user has not revoked access
+     * @param dest Address to which the call will be sent
+     * @param howToCall Which kind of call to make
+     * @param calldata Calldata to send
+     * @return Result of the call (success or failure)
+     */
+    function proxy(
+        address dest,
+        HowToCall howToCall,
+        bytes calldata
```

Comment:
calldata is data location type, not name of variable
https://docs.soliditylang.org/fr/latest/types.html?highlight=calldata#data-location-and-assignment-behaviour


replace with something like :

`bytes calldata data`

and 8 lines below

`result = dest.delegatecall(data);`